File: tests/benchmark/corpus/go/safe/safe_dominated.go
package main

import (
	"net/http"
	"os/exec"
)

var allowed = map[string]bool{"ls": true, "pwd": true}

func handler(w http.ResponseWriter, r *http.Request) {
	cmd := r.URL.Query().Get("cmd")
	if !allowed[cmd] {
		http.Error(w, "denied", 403)
		return
	}
	exec.Command(cmd).Run()
}

File: tests/benchmark/corpus/javascript/safe/safe_dominated.js
const express = require('express');
const { exec } = require('child_process');
const app = express();
const ALLOWED = ['ls', 'pwd'];
app.get('/run', (req, res) => {
    const cmd = req.query.cmd;
    if (!ALLOWED.includes(cmd)) { return res.status(403).send('denied'); }
    exec(cmd);
});
